test/test_diagnostics.jl

```diff
@@ -12,32 +12,49 @@ function nan_checker_aborts_simulation(arch)
     return nothing
 end
 
-TestModel(::GPU, FT, ν=1.0, Δx=0.5) =
+TestModel_VerticallyStrectedRectGrid(arch, FT, ν=1.0, Δx=0.5) =
     IncompressibleModel(
-          grid = RegularRectilinearGrid(FT, size=(3, 3, 3), extent=(3Δx, 3Δx, 3Δx)),
+          grid = VerticallyStretchedRectilinearGrid(FT, size=(3, 3, 3), x=(0, 3Δx), y=(0, 3Δx), zF=0:Δx:3Δx,),
```

Comment:
```suggestion
          grid = VerticallyStretchedRectilinearGrid(FT, architecture = arch, size=(3, 3, 3), x=(0, 3Δx), y=(0, 3Δx), zF=0:Δx:3Δx,),
```

Reply:
@tomchor I accidently made a commit that I later realized was wrong so I reverted it.

I think the test was failing because the stretched grid needed a `architecture = arch` kwarg.

Maybe this is a reason to avoid having default architectures. Or at least we could check that the grid and model architectures are consistent as part of #1453.

Reply:
That makes sense. Thanks!